packages/kotti-ui/source/kotti-field-file-upload/validators.ts

```diff
@@ -0,0 +1,50 @@
+import { KottiFieldFileUpload } from './types'
+
+export const isSelectingMultipleFilesWhenNotAllowed = (
+	allowMultiple: boolean,
+	selectionSize: number,
+): boolean => !allowMultiple && selectionSize > 1
+
+const isValidFileSize = (fileSize: number, maxFileSize: number): boolean =>
+	fileSize < maxFileSize
+
+const isValidFileExtension = (
+	fileName: string,
+	validFileExtensions: Array<string>,
+): boolean => {
+	if (validFileExtensions.length === 0) return true
+
+	const fileExtension = (fileName.split('.').pop() ?? '').trim().toUpperCase()
```

Comment:
For future reference
`string.prototype.localCompare` could be useful with its options tweaked, because it can ignore casing and punctation (which includes spaces)
check the docs for it

I tried this example and they both return `0` which means they are equal

```
'jpg.'.localeCompare(', JPG', undefined, {ignorePunctuation: true, sensitivity: "accent"})
```

Reply:
in that case you'd just
```
const extension = fileName.split('.').pop() ?? null
if(extension === null) return false

return validFileExtensions.some(e => e.localeCompare(extension, undefined, {ignorePunctuation: true, sensitivity: "accent"}) === 0)
```